pipeline/frontend/yaml/compiler/compiler.go

```diff
@@ -136,6 +136,14 @@ func (c *Compiler) Compile(conf *yaml.Config) *backend.Config {
 	var stage *backend.Stage
 	var group string
 	for i, container := range conf.Pipeline.Containers {
+		//Skip if local and should not run local
+		if c.local && !container.Constraints.Local.Bool() {
+			continue
+		}
+		//Skip if remote and should only run local
+		if !c.local && container.Constraints.Local.Bool() {
+			continue
```

Comment:
we didn't have this block previously. Have you tested how it will work? because I think `Bool()` is true by default, which means this would prevent all steps from running on the server